src/libs/OptionsListUtils.ts

```diff
@@ -964,7 +964,12 @@ function getCategoryListSections(
     }
 
     if (searchInputValue) {
-        const searchCategories = enabledCategories.filter((category) => category.name.toLowerCase().includes(searchInputValue.toLowerCase()));
+        const searchCategories = enabledCategories
+            .filter((category) => category.name.toLowerCase().includes(searchInputValue.toLowerCase()))
+            .map((category) => ({
+                ...category,
+                isSelected: selectedOptions.some((selectedOption) => selectedOption.name === category.name),
```

Comment:
@dukenv0307 @puneetlath i think we should optimize this. This is a `O(N^2)` call for every search input.

Reply:
We have a list of selected options and a list unselected, why rebuild the list?

Reply:
Yeah, ideally there is a more performant way we could do this.

Reply:
@dukenv0307 we don't need to recalculate the selected options on every keypress. 

eg: we are calculating the enabled categories only once over here - https://github.com/Expensify/App/pull/37479/files#diff-7d43f3c3e4decdb0057928e895f456d51c0f4081837edd5e2c4ed7ddd570e274R947

Reply:
ah okay, thanks!